htmlhint-server/src/server.ts

```diff
@@ -60,6 +65,74 @@ function getRange(error: htmlhint.Error, lines: string[]): any {
     };
 }
 
+function loadRules(HTMLHint: any, force: boolean): any {
+    if (force) {
+        rulesLoaded = false;
+    }
+
+    if (rulesLoaded || htmlhint == null) {
+        return;
+    }
+    try {
+        if (!settings.htmlhint || !settings.htmlhint.rulesDir) {
+            return;
+        }
+        let rulesDir = settings.htmlhint.rulesDir;
+        let absoluteDir = '';
+        //check absolute path
+        let exists = fs.existsSync(rulesDir);
+        if (exists) {
+            absoluteDir = rulesDir;
+        } else {
+            //relative dir, find it based on workspace roots
+            absoluteDir = path.resolve(rootFolder, rulesDir);
+            exists = fs.existsSync(absoluteDir);
+        }
+
+        if (exists) {
+            //load all rules
+            loadCustomRules(absoluteDir, HTMLHint);
+        }
+    }
+    catch (e) { }
+    finally {
+        rulesLoaded = true;
+    }
+}
+
+// load custom rles
+function loadCustomRules(rulesdir:string, HTMLHint:any):any{
+    rulesdir = rulesdir.replace(/\\/g, '/');
+    if(fs.existsSync(rulesdir)){
+        if(fs.statSync(rulesdir).isDirectory()){
+            rulesdir += /\/$/.test(rulesdir)?'':'/';
+            rulesdir += '**/*.js';
+            var arrFiles = glob.sync(rulesdir, {
+                'dot': false,
+                'nodir': true,
+                'strict': false,
+                'silent': true
+            });
+            arrFiles.forEach(function(file){
+                loadRule(file, HTMLHint);
```

Comment:
What's the motivation for letting someone specify a directory here?  It seems like doing that will result in the extension require-ing every JS file in the directory, which could easily be problematic.  

Can you support a single file, or an explicit list of files?  
What does basic htmlhint allow here?

Reply:
The way the plugin works (as I understand it) is you can create multiple rules in as many files and subfolders as you wish.
Each .js file in the rules folder will be loaded into memory.

Reply:
I thought they had to be specified in the rules list as well via htmlhintrc or  --rules on the command line.